sql/2020/04_Fonts/04_07.font_resource_hints_with_fcp.sql

```diff
@@ -0,0 +1,84 @@
+#standardSQL
+#font_resource_hints_with_fcp(??NoResult)
+CREATE TEMPORARY FUNCTION getResourceHints(payload STRING)
+RETURNS ARRAY < STRUCT < name STRING, href STRING >>
+    LANGUAGE js AS '''
+var hints = new Set(['preload', 'prefetch', 'preconnect', 'prerender', 'dns-prefetch']);
+try {
+    var $ = JSON.parse(payload);
+    var almanac = JSON.parse($._almanac);
+    return almanac['link-nodes'].nodes.reduce((results, link) => {
+        var hint = link.rel.toLowerCase();
+        if (!hints.has(hint)) {
+            return results;
+        }
+        results.push({
+            name: hint,
+            href: link.href
+        });
+        return results;
+    }, []);
+} catch (e) {
+    return [null];
+}
+''';
+SELECT
+  client,
+  name,
+  COUNT(DISTINCT page) AS freq_hints,
+  total_page,
+  COUNT(DISTINCT page)*100/total_page AS pct_hints,
+  COUNTIF(fast_fcp>=0.75)*100/COUNT(0) AS pct_good_fcp_hints,
+  COUNTIF(NOT(slow_fcp>=0.25)
+    AND NOT(fast_fcp>=0.75))*100/COUNT(0) AS pct_ni_fcp_hints,
+  COUNTIF(slow_fcp>=0.25)*100/COUNT(0) AS pct_poor_fcp_hints,
+FROM (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    url AS page,
+    hint.name,
+    hint.href AS hint_url
+  FROM
+    `httparchive.pages.2020_08_01_*`
+    LEFT JOIN UNNEST(getResourceHints(payload)) AS hint)
+LEFT JOIN (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    url AS page,
+    type
+  FROM
+    `httparchive.summary_requests.2020_08_01_*`
+  WHERE
+    type='font')
+USING
+  (client, page)
+JOIN (
+  SELECT DISTINCT
+    origin, device,
+    fast_fcp,
+    slow_fcp,
+  FROM
+    `chrome-ux-report.materialized.device_summary`
+  WHERE
+    yyyymm=202008)
```

Comment:
The `device_summary` table is inconsistent with the `country_summary` table in that it's partitioning field is actually `date` rather than `yyyymm`. Sorry for the confusion!

So this query can be optimized:
```suggestion
    date='2020-08-01')
```

Reply:
sorry to request the time on review again, as I couldn't find a road without 0 result after running revised query. Thank you.

Reply:
It would be weird if the same query was producing results for me but not for you. Are you sure you're running [the latest version](https://github.com/HTTPArchive/almanac.httparchive.org/blob/Fonts-2020-queries/sql/2020/04_Fonts/04_07.font_resource_hints_with_fcp.sql)? Is it possible you've run out of quota? Let me know if there are any error messages related to quota or anything else.